FileShareService.DesktopClient.Core/Jobs/NewBatchJob.cs

```diff
@@ -52,11 +53,59 @@ public void Validate(JToken jsonToken)
             }
 
             //Check for files
-            if (jsonToken.SelectToken("actionParams.files") != null && 
+            if (jsonToken.SelectToken("actionParams.files") != null &&
                 jsonToken.SelectToken("actionParams.files").Type != JTokenType.Array)
             {
                 ErrorMessages.Add($"Invalid file object.");
             }
+
+            //Check for file attributes
+
+            foreach (JToken fileObj in jsonToken.SelectToken("actionParams.files"))
+            {
+                JToken? fileAttributeToken = fileObj.SelectToken("attributes");
+                var searchPath = fileObj.SelectToken("searchPath");
+
+                if (fileAttributeToken != null && JArray.FromObject(fileAttributeToken).Count > 0)
+                {
+                    if (fileAttributeToken?.Type != JTokenType.Array)
+                    {
+                        ErrorMessages.Add("Invalid file attribute.");
+                    }
+                    else if (fileAttributeToken.HasValues)
+                    {
+                        //Check for batch attribute key and value
+                        foreach (JToken? fileAttribute in fileAttributeToken)
```

Comment:
Do you think it would be possible to refactor the duplicated code for validating batch attributes and file attributes into a single method?

Reply:
The reasons for not creating a common function for file and batch attribute-
file attributes can be available for each file unlike batch attribute.
Also there are more checks for file attribute than batch attribute for eg: "File attribute key/value cannot be blank" and searchpath appended to error message to identify error occurs for which particular file. Can you please suggest

Reply:
Ok, I agree. It would be nice to have a single method for validating attributes, but I can see the validation criteria for batch attributes is different to the validation criteria for file attributes, and an additional complication is the error messages would have to be passed in so I'm happy to resolve this with no action needed